python/pyarrow/lib_gpu.pyx

```diff
@@ -0,0 +1,809 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import six
+from pyarrow.compat import tobytes
+from pyarrow.lib cimport *
+from pyarrow.includes.libarrow_gpu cimport *
+from pyarrow.lib import py_buffer, allocate_buffer
+cimport cpython as cp
+
+cdef class CudaDeviceManager:
```

Comment:
This is entirely a singleton in C++, so I don't think it makes sense to expose this as a Python class. Furthermore, you already have top-level functions such as `cuda_allocate_host_buffer`...